File: abc/abc172/abc172_d.cs
// https://atcoder.jp/contests/abc172/submissions/14784523

using System;

class Program
{
    static void Main()
    {
        var N = long.Parse(Console.ReadLine());

        var sum = 0L;
        for (int i = 1; i < N + 1; i++)
        {
            var q = N / i;
            sum += q * (q + 1) * i / 2;
        }
        Console.WriteLine(sum);
    }
}

File: abc/abc172/abc172_d.fs
// https://atcoder.jp/contests/abc172/submissions/14784108

let N = stdin.ReadLine() |> int64

let rec solver i sum =
    if i = N + 1L then sum
    else
        let q = N / i
        solver (i + 1L) (sum + q * (q + 1L) * i / 2L)

solver 1L 0L |> stdout.WriteLine
